mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest.go

```diff
@@ -257,6 +256,24 @@ func (r *EncryptionAtRestRS) Create(ctx context.Context, req resource.CreateRequ
 	}
 }
 
+func resourceMongoDBAtlasEncryptionAtRestCreateRefreshFunc(ctx context.Context, projectID string, conn *matlas.Client, encryptionAtRestReq *matlas.EncryptionAtRest) retry.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		encryptionResp, _, err := conn.EncryptionsAtRest.Create(ctx, encryptionAtRestReq)
+		if err != nil {
+			if strings.Contains(err.Error(), "CANNOT_ASSUME_ROLE") || strings.Contains(err.Error(), "INVALID_AWS_CREDENTIALS") ||
```

Comment:
This code has just been moved from its previous place. As per one [previous comment](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1463) I am not sure we can use here `errors.Is(...)` statement because it would not be the same as checking if string contains

Reply:
You can define two errors

```
var cannotAssumeRole := errors.new("CANNOT_ASSUME_ROLE") 
var invalidAWSCredentials "= errors.new("INVALID_AWS_CREDENTIALS")
```
and the use `errors.Is(err, cannotAssumeRole)`

Reply:
@andreaangiolillo I know we can do that, but my point here is that I don't know if `err.Error()` **only** contains that exact string or it contains more. Does it make sense or am I missing something?

Based on the `errors.Is()` documentation:
```
// Is reports whether any error in err's tree **matches** target.
```

Reply:
ah, thanks for the clarification. You can find out in mms 

> CANNOT_ASSUME_ROLE(
>       "Atlas cannot assume the specified role.", "Atlas cannot assume the specified role (%s)."),

>   INVALID_AWS_CREDENTIALS(HttpServletResponse.SC_BAD_REQUEST, "Invalid AWS credentials."),


https://github.com/10gen/mms/blob/74bd3430d12c4c446b8f214222b2c3ea9e91a452/server/src/main/com/xgen/svc/mms/api/res/atlas/ApiAtlasEncryptionAtRestResource.java#L273

https://github.com/10gen/mms/blob/74bd3430d12c4c446b8f214222b2c3ea9e91a452/server/src/main/com/xgen/svc/mms/api/res/atlas/ApiAtlasEncryptionAtRestResource.java#L406